apps/communications/contacts/js/activities.js

```diff
@@ -77,13 +77,29 @@ var ActivityHandler = {
     return this.currentlyHandling && list.indexOf(this.activityName) === -1;
   },
 
+  checkCancelableActivity: function() {
+    if (this.currentlyHandling) {
+      var alternativeTitle = null;
+      var activityName = this.activityName;
+      if (activityName === 'pick' || activityName === 'update') {
+        alternativeTitle = 'selectContact';
+      }
+      var groupsList = document.getElementById('groups-list');
+      HeaderUI.setSelectModeClass(groupsList, activityName,
+                              this.activityDataType);
+      HeaderUI.setupCancelableHeader(alternativeTitle);
+    } else {
+      HeaderUI.setupActionableHeader();
+    }
```

Comment:
Do we need to mix UI and Activities? Why dont we have something as `ActivityHandler.isCancelable().then(function(isCancelable) {..... // UI Code})`?